src/utils/ExportData.js

```diff
@@ -179,19 +235,19 @@ const saveFile = (data, openErrorDialog) => {
   }
 };
 
-const createGraphML = (networkData, openErrorDialog) => {
+const createGraphML = (networkData, variableRegistry, openErrorDialog) => {
   // default graph structure
   const xml = setUpXml();
   const graph = xml.getElementsByTagName('graph')[0];
   const graphML = xml.getElementsByTagName('graphml')[0];
 
   // generate keys for attributes
-  generateKeys(graph, graphML, networkData.nodes, 'node', ['id']);
-  generateKeys(graph, graphML, networkData.edges, 'edge', ['from', 'to', 'id']);
+  generateKeys(graph, graphML, networkData.nodes, 'node', ['id'], variableRegistry);
+  generateKeys(graph, graphML, networkData.edges, 'edge', ['from', 'to', 'id'], variableRegistry);
 
   // add nodes and edges to graph
-  addElements(graph, graphML.namespaceURI, networkData.nodes, 'node', ['id']);
-  addElements(graph, graphML.namespaceURI, networkData.edges, 'edge', ['from', 'to', 'id'], true);
+  addElements(graph, graphML.namespaceURI, networkData.nodes, 'node', ['id'], variableRegistry);
+  addElements(graph, graphML.namespaceURI, networkData.edges, 'edge', ['from', 'to', 'id'], variableRegistry, true);
 
   saveFile(xmlToString(xml), openErrorDialog);
```

Comment:
I know this is out of scope for this PR, but while we are updating this, I would really like it if you could move the file handling code out of this method and into a utility function elsewhere. This would be in preparation for other export functionality being able to reuse the file handling/writing.

So perhaps create methods in `./utils` for writing files that continue to use system dialogs on electron and share on Cordova. The interface for these can then handle, via callbacks or whatever, integration with the dialog system.

Then the actual file encoding methods here would just return strings that would then be passed to these file handling methods.

Reply:
I'll take a look at this and some testing as well.